platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -1168,14 +1615,95 @@ namespace platoon_strategic_ihp
             return MobilityRequestResponse::NO_RESPONSE;
         }
     }
+    
+    // UCLA: add request call-back function for lead with operation state (for cut-in join)
+    MobilityRequestResponse PlatoonStrategicIHPPlugin::mob_req_cb_leadwithoperation(const cav_msgs::MobilityRequest& msg)
+    {
+        /*
+        *   Current leader leading while opening the gap for cut-in join.
+        */
+
+        // Check request plan type  
+        cav_msgs::PlanType plan_type = msg.plan_type;
+        if (plan_type.type == cav_msgs::PlanType::PLATOON_CUTIN_JOIN) 
+        {
+            // task 1. slow down to create gap 
+            int join_index = msg.join_index;
+            // determine if joining vehicle in position
+            double cutinDtdDifference = current_downtrack_ - pm_.platoon[0].vehiclePosition - config_.vehicleLength;
+            bool isFrontJoinerInPosition = (cutinDtdDifference <= 1.5*config_.vehicleLength) || (cutinDtdDifference >= -1.5*config_.vehicleLength);
```

Comment:
General joining philosophy question: we can join a platoon from many car lengths away, right (e.g. a rear join)?  Can't we also join from many car lengths in front of the platoon, by just having the joiner gradually slow down to close the gap?  If these are true, then why does a cut-in need to be almost exactly at the front of the platoon whereas a same-lane front join may be from a lot farther in front?